lib/config-loader.js

```diff
@@ -290,6 +289,12 @@ function mergeArrays(target, config, keyPrefix) {
   return null; // no error
 }
 
+function mergeArraysDiffLength(target, config) {
+  var newTarget = _.cloneDeep(target, true);
+  Array.prototype.splice.apply(target, [0, target.length].concat(_.union(newTarget, config)));
```

Comment:
Hi, this line is doing a lot of work and it's difficult to understand, at least to me. Could you please refactor the code and/or add comments to explain better what's going on?